packages/core-plugin-api/src/extensions/extensions.tsx

```diff
@@ -245,7 +246,11 @@ export function createReactExtension<
                   ...(mountPoint && { routeRef: mountPoint.id }),
                 }}
               >
-                <Component {...props} />
+                <PluginOptionsProvider
+                  pluginOptions={plugin.getPluginOptions()}
+                >
+                  <Component {...props} />
+                </PluginOptionsProvider>
```

Comment:
Let's call this `PluginProvider` instead for now, because I believe we'll re-use it for #10256. Gonna keep it experimental either way so we can see where it lands. Pass in the entire `plugin` instance instead of just the options, and handle the unpacking within the provider.